src/main/scala/beam/sim/BeamWarmStart.scala

```diff
@@ -2,98 +2,78 @@ package beam.sim
     }
   }
 
-  def read: Option[TravelTime] = {
-    if (isWarmMode) {
-      getWarmStartFilePath("linkstats.csv.gz", rootFirst = false) match {
-        case Some(statsPath) =>
-          if (Files.exists(Paths.get(statsPath))) {
-            val travelTime = getTravelTime(statsPath)
-            logger.info("Read travel times from {}.", statsPath)
-            Some(travelTime)
-          } else {
-            logger.warn("Travel times failed to warm start, stats not found at path ( {} )", statsPath)
-            None
-          }
-        case None =>
-          logger.warn("Travel times failed to warm start, stats not found at path ( {} )", srcPath)
+  def readTravelTime: Option[TravelTime] = {
+    getWarmStartFilePath("linkstats.csv.gz", rootFirst = false) match {
+      case Some(statsPath) =>
+        if (Files.isRegularFile(Paths.get(statsPath))) {
+          val travelTime = getTravelTime(statsPath)
+          logger.info("Read travel times from {}.", statsPath)
+          Some(travelTime)
+        } else {
+          logger.warn("Travel times failed to warm start, stats not found at path ( {} )", statsPath)
           None
-      }
-    } else {
-      None
+        }
+      case None =>
+        logger.warn("Travel times failed to warm start, stats not found at path ( {} )", srcPath)
+        None
     }
   }
 
-  /**
-    * initialize population.
-    */
-  def warmStartPopulation(matsimConfig: Config): Unit = {
-    if (isWarmMode) {
-      getWarmStartFilePath("plans.xml.gz") match {
-        case Some(statsPath) =>
-          if (Files.exists(Paths.get(statsPath))) {
-            val file = loadPopulation(parentRunPath, statsPath)
-            matsimConfig.plans().setInputFile(file)
-            logger.info("Population successfully warm started from {}", statsPath)
-          } else {
-            logger.warn(
-              "Population failed to warm start, plans not found at path ( {} )",
-              statsPath
-            )
-          }
-        case None =>
-          logger.warn(
-            "Population failed to warm start, plans not found at path ( {} )",
-            srcPath
-          )
-      }
+  private def warmstartFile(description: String, fileName: String): String = {
+    getWarmStartFilePath(fileName) match {
+      case Some(compressedFile) =>
+        if (Files.isRegularFile(Paths.get(compressedFile))) {
+          loadPopulation(parentRunPath, compressedFile)
+        } else {
+          throwErrorFileNotFound(description, compressedFile)
+        }
+      case None =>
+        throwErrorFileNotFound(description, srcPath)
     }
   }
 
-  private def loadPopulation(runPath: String, populationFile: String): String = {
-    val plansPath = Paths.get(runPath, "warmstart_plans.xml")
-    unGunzipFile(populationFile, plansPath.toString, false)
-    plansPath.toString
+  private def throwErrorFileNotFound(fileDesc: String, path: String): String = {
```

Comment:
There is (FileNotFoundException)[https://docs.oracle.com/javase/7/docs/api/java/io/FileNotFoundException.html] in Java. Why don't use it? At least the name of the method contains file not found

Reply:
done